modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/DeployEnvironments.java

```diff
@@ -113,10 +117,25 @@ private Map<String, String> getEnv(Map<String, Object> serviceRequestProperties,
                                 ? aesUtil.decode(variable.getValue()) : variable.getValue());
             }
         }
-
+        setProviderOsAuthUrl(csp, variables);
         return variables;
     }
 
+    private void setProviderOsAuthUrl(Csp csp, Map<String, String> variables) {

```

Comment:
this logic must be part of plugin of each cloud provider.
the Orchestrator must only ask the corresponding plugin for these variables and the translation must come from it.